Lib/asyncio/proactor_events.py

```diff
@@ -232,8 +232,9 @@ class _ProactorBaseWritePipeTransport(_ProactorBasePipeTransport,
 
     def write(self, data):
         if not isinstance(data, (bytes, bytearray, memoryview)):
-            raise TypeError('data argument must be byte-ish (%r)',
-                            type(data))
+            msg = ("data argument must be a bytes-like object, not '%s'" %
```

Comment:
`%r` looks OK to me. Is there a reason to use `'%s'` with `type(data).__name__`?